graspologic/embed/ase.py

```diff
@@ -158,4 +160,63 @@ def fit(self, graph, y=None):
 
         self.n_features_in_ = len(A)
         self._reduce_dim(A)
+
+        # for out-of-sample
+        inv_eigs = np.diag(1 / self.singular_values_)
+        self.pinv_left_ = self.latent_left_ @ inv_eigs
+        if self.latent_right_ is not None:
+            self.pinv_right_ = self.latent_right_ @ inv_eigs
+
+        self.is_fitted_ = True
         return self
+
+    def predict(self, y):
+        """
+        Obtain out-of-sample latent positions for vertices not in the original embedding.
+        For more details, see [1].
+
+        Parameters
+        ----------
+        y : array_like or tuple, shape (n_oos_vertices, n_vertices)
+            out-of-sample matrix.
+            If tuple, graph is directed and y[0] contains edges from y to other nodes.
+
+        Returns
+        -------
+        array_like or tuple, shape (n_oos_vertices, n_components)
+            Out-of-sample prediction for the latent position(s) of y.
+            If undirected, returns array.
+            If directed, returns (y_left, y_right).
+
+        References
+        ----------
+        .. [1] Levin, K., Roosta-Khorasani, F., Mahoney, M. W., & Priebe, C. E. (2018).
+        Out-of-sample extension of graph adjacency spectral embedding. PMLR: Proceedings
+        of Machine Learning Research, 80, 2975-2984.
```

Comment:
just a note (should revisit) we've mostly kept references to classes thought I see why it makes sense here in this case.

Reply:
I'd argue in favor of leaving the reference in, just because of the very specific use case. I could move the reference to the class docstring, though

Reply:
that is what i meant, defs don't remove it.